cmd/node/factory/interface.go

```diff
@@ -14,9 +14,10 @@ type HeaderSigVerifierHandler interface {
 	IsInterfaceNil() bool
 }
 
-// HeaderIntegrityVerifierHandler is the interface needed to check that a header's integrity is correct
-type HeaderIntegrityVerifierHandler interface {
+// HeaderVersioningHandler encapsulates methods useful to check that a header's versioning is correct
+type HeaderVersioningHandler interface {
```

Comment:
it still has the integration verification methods :) the interface name could have remained the same.

Reply:
It also provides the correct version now. That is why I have renamed it to handler as it manages the software versions for headers. :)

Reply:
And the chain Id. Could have remained like before